### PR TITLE
clean up unsupport etcd version

### PR DESCRIPTION
**Pull request by kkkkun**
#### What type of PR is this?
/kind cleanup

#### Which issue(s) this PR fixes:

Fixes # https://github.com/kubernetes/kubernetes/issues/117906

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: cluster/images/etcd/Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@
 # Except from etcd-$(version) and etcdctl-$(version) binaries, we also
 # need etcd and etcdctl binaries for backward compatibility reasons.
 # That binary will be set to the last version from $(BUNDLED_ETCD_VERSIONS).
-BUNDLED_ETCD_VERSIONS?=3.0.17 3.1.20 3.2.32 3.3.17 3.4.18 3.5.8
```
[the remark this answers is not in the document]
Comment (kkkkun) [reply cluster/images/etcd/Makefile:29]: > don't we have to keep `3.3.x` ?

According https://github.com/kubernetes/kubernetes/pull/117332#discussion_r1167300954, current clientv3 only support etcd 3.4 and 3.5

File: cmd/kubeadm/app/constants/constants.go [hunk 2]
```diff
@@ -467,15 +467,11 @@ var (
...
 	SupportedEtcdVersion = map[uint8]string{
-		13: "3.2.24",
-		14: "3.3.10",
-		15: "3.3.10",
-		16: "3.3.17-0",
 		17: "3.4.3-0",
 		18: "3.4.3-0",
-		19: "3.4.13-0",
-		20: "3.4.13-0",
-		21: "3.4.13-0",
```
Comment (kkkkun) [inline cmd/kubeadm/app/constants/constants.go:478]: Consistent with https://github.com/kubernetes/kubernetes/blob/e1ad9bee5bba8fbe85a6bf6201379ce8b1a611b1/cluster/images/etcd/migrate/options.go#L31

**Review by kkkkun — COMMENTED** [review]
Add comments

File: cmd/kubeadm/app/constants/constants.go [hunk 3]
```diff
@@ -467,15 +467,11 @@ var (
...
+		19: "3.4.18-0",
+		20: "3.4.18-0",
+		21: "3.4.18-0",
```
Comment (kkkkun) [inline cmd/kubeadm/app/constants/constants.go:474]: Keep consistent with https://github.com/kubernetes/kubernetes/blob/e1ad9bee5bba8fbe85a6bf6201379ce8b1a611b1/cluster/images/etcd/migrate/options.go#L31

File: cmd/kubeadm/app/constants/constants.go [hunk 4]
```diff
@@ -467,15 +467,11 @@ var (
 
 	// SupportedEtcdVersion lists officially supported etcd versions with corresponding Kubernetes releases
 	SupportedEtcdVersion = map[uint8]string{
-		13: "3.2.24",
-		14: "3.3.10",
-		15: "3.3.10",
-		16: "3.3.17-0",
```
[the remark this answers is not in the document]
Comment (neolit123) [reply cmd/kubeadm/app/constants/constants.go:473]: we should drop < 3.4?

**Review by neolit123 — COMMENTED** [review]
/approve

**Comment by jpbetz** [comment]
/lgtm
/approve
For cluster/images changes

Please cancel hold once kubeadm changes are ready

**Review by SataQiu — APPROVED** [review]
/lgtm

**Comment by pacoxu** [comment]
> /hold
> /assign @jpbetz

Since @jpbetz approved it,
/unhold

File: cmd/kubeadm/app/constants/constants.go (hunk 3 above)
Comment (wzshiming) [reply cmd/kubeadm/app/constants/constants.go:474]: I found that this tag does not exist at the moment, maybe someone needs to sync it
`registry.k8s.io/etcd:3.4.18-0`

Comment (pacoxu) [reply cmd/kubeadm/app/constants/constants.go:474]: @neolit123 not sure if we need to revert the version map here of change  `3.4.13-0` to `3.4.18-0`.

3.4.18 is the last patch of 3.4.x etcd.

But we did not build it in kube. The latest one is `3.4.13-3`.
https://github.com/kubernetes/kubernetes/blob/release-1.21/cluster/images/etcd/Makefile#L32-L37

```
[root@pacoall ~]# docker pull registry.k8s.io/etcd:3.4.13-4
Trying to pull repository registry.k8s.io/etcd ... 
Pulling repository registry.k8s.io/etcd
Error: image etcd:3.4.13-4 not found
[root@pacoall ~]# docker pull registry.k8s.io/etcd:3.4.13-3
Trying to pull repository registry.k8s.io/etcd ... 
3.4.13-3: Pulling from registry.k8s.io/etcd
5749e56bea71: Pull complete 
99ef90998473: Pull complete 
bc941152f6e4: Pull complete 
011c6af5a966: Pull complete 
2e72bd25dd5b: Pull complete 
Digest: sha256:528ff5e3c80ca0d58d1ab15dc329dc300f9c3fefaffa4c822ca2c389ce254644
Status: Downloaded newer image for registry.k8s.io/etcd:3.4.13-3
```

Comment (pacoxu) [reply cmd/kubeadm/app/constants/constants.go:474]: @jpbetz @serathius

https://github.com/kubernetes/kubernetes/blob/1ff1a26426fedbb929bc966cebb9f90995e52725/cluster/images/etcd/Makefile#L29-L40

Should we build all `BUNDLED_ETCD_VERSIONS` in some ways?

Comment (kkkkun) [reply cmd/kubeadm/app/constants/constants.go:474]: It does not have `3.4.18-x`. https://github.com/kubernetes/k8s.io/blob/main/registry.k8s.io/images/k8s-staging-etcd/images.yaml#LL5C85-L5C85

May be we should revise to `3.4.13` ?

Comment (neolit123) [reply cmd/kubeadm/app/constants/constants.go:474]: > not sure if we need to revert the version map here of change `3.4.13-0` to `3.4.18-0`.

we have to, if the .18 image does not exist.
only once an image exists, we can use it in k/k code.

Comment (kkkkun) [reply cmd/kubeadm/app/constants/constants.go:474]: @xmudrii Would you please look this?

Comment (neolit123) [reply cmd/kubeadm/app/constants/constants.go:474]: @pacoxu i submitted PR https://github.com/kubernetes/kubernetes/pull/118733 for kubeadm

**Pull request merged**